### PR TITLE
Performance boost for SDL ASCII drawing using color modulated textures

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary

 ```SUMMARY: Performance "Performance boost for SDL drawing using color modulated textures"```

#### Purpose of change

- increase performance of SDL ASCII drawing;
- allow selection of different renderers (some renderers can increase performance on a given system);
- allow using of [ZoomIt](https://docs.microsoft.com/en-us/sysinternals/downloads/zoomit) (and possibly other similar software) with proper renderer.

#### Describe the solution

- replaced `SDL_RenderFillRect` with `SDL_RenderCopy` of a color modulated texture (fallback to `SDL_RenderFillRect` when color modulation is unsupported);
- added `USE_COLOR_MODULATED_TEXTURES` graphic option which will enable usage of color modulated texture of color modulation (if it is supported); 
- added `RENDERER` graphic option which allows selecting which render driver to use (`direct3d`, `opengl`, `opengles` or `software`);
- added output of draw benchmark to `debug.log` and allowed selection of benchmark length.

#### Additional context

image

Also see #19165 and #21748.

### CODE REVIEW

File: src/sdltiles.cpp [hunk 1]
```diff
@@ -348,11 +397,31 @@ void WinCreate()
     format.reset( SDL_AllocFormat( wformat ) );
     throwErrorIf( !format, "SDL_AllocFormat failed" );
 
-    bool software_renderer = get_option<bool>( "SOFTWARE_RENDERING" );
+    bool software_renderer = get_option<bool>( "SOFTWARE_RENDERING" ) || get_option<std::string>( "RENDERER" ).empty();
+    int renderer_id = -1;
+    std::string renderer_name;
+    if( software_renderer )
```
Comment (kevingranade) [inline src/sdltiles.cpp]: Broken brace.

File: src/options.cpp [hunk 2]
```diff
@@ -1514,6 +1514,17 @@ void options_manager::add_options_graphics()
...
+#if (defined TILES)
+         // No renderer selection in non-TILES mode
+         cata_tiles::build_renderer_list(),
+#else
+    {   { "software", translate_marker( "software" ) }
+    },
+#endif
+         "software", COPT_NO_HIDE );
+
     add( "SOFTWARE_RENDERING", "graphics", translate_marker( "Software rendering" ),
```
Comment (kevingranade) [inline src/options.cpp]: We should get rid of this option since it's redundant with renderer=software

Comment (ZhilkinSerg) [reply src/options.cpp]: Done.

**Comment by ZhilkinSerg** [comment]
Yes, I forgot to handle:

```CPP
get_option( "FRAMEBUFFER_ACCEL" ).setPrerequisite( "SOFTWARE_RENDERING" );
```

Should I just remove the prerequisite here or should I expand `setPrerequisite` function to handle non-boolean values?

E.g.:

```CPP
get_option( "FRAMEBUFFER_ACCEL" ).setPrerequisite( "RENDERER", "software" );
```
and/or
```CPP
get_option( "FRAMEBUFFER_ACCEL" ).setPrerequisite( "RENDERER", { "software", "directx" } );
```

**Pull request merged**